src/YieldHandler.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.14;
+
+import "./ITopUpHandler.sol";
+import "vault-interfaces/src/DataTypes.sol";
+import "vault-interfaces/src/ICauldron.sol";
+import "vault-interfaces/src/ILadle.sol";
+import "yield-utils-v2/contracts/cast/CastU256I128.sol";
+import "yield-utils-v2/contracts/token/IERC20.sol";
+
+interface ICauldronCustom {
+    function level(bytes12 vaultId) external view returns (int256);
+}
+
+interface ILadleCustom {
+    function addModule(address module, bool set) external;
+
+    function moduleCall(address module, bytes calldata data) external payable returns (bytes memory result);
+}
+
+interface IHealerModule {
+    function heal(bytes12 vaultId_, int128 ink, int128 art) external payable;
+}
+
+contract YieldHandler is ITopUpHandler {
+    using CastU256I128 for uint256;
+
+    DataTypes.Vault vault;
+    ICauldron cauldron;
+    IHealerModule healer;
+    IJoin join;
+    ILadle ladle;
+
+    constructor(ICauldron cauldron_, IHealerModule healer_, ILadle ladle_) {
```

Comment:
There is a kind of unofficial ordering to the parameters, in which each contract is stated in the order in which they would be deployed. So it needs to be Cauldron, Ladle, HealerModule.

Please follow the same ordering when declaring state variables above.